sdk/servicebus/Microsoft.Azure.ServiceBus/src/IQueueClient.cs

```diff
@@ -77,5 +77,13 @@ public interface IQueueClient : IReceiverClient, ISenderClient
         /// <param name="sessionHandlerOptions">Options used to configure the settings of the session pump.</param>
         /// <remarks>Enable prefetch to speed up the receive rate. </remarks>
         void RegisterSessionHandler(Func<IMessageSession, Message, CancellationToken, Task> handler, SessionHandlerOptions sessionHandlerOptions);
+
+        /// <summary>
+        /// Unregister session handler from the receiver if there is an active session handler registered. This operation waits for the completion
+        /// of inflight receive and session handling operations to finish and unregisters future receives on the session handler which previously 
+        /// registered. 
+        /// <param name="inflightSessionHandlerTasksWaitTimeout"> is the waitTimeout for inflight session handling tasks.  
+        /// </summary>
+        Task UnregisterSessionHandlerAsync(TimeSpan inflightSessionHandlerTasksWaitTimeout);
```

Comment:
Should we add another function signature for UnregisterSession/MessageHandlerAsync that takes in empty parameter and set a default wait timeout?

Reply:
you could change the function signature like this and check for null in the code
Task UnregisterSessionHandlerAsync(TimeSpan? inflightSessionHandlerTasksWaitTimeout = null);

What should null represent though?  It can't be zero or draining will not happen so infinite time seems like the best default.

Reply:
Yes, I thought about this but our current function signature doesn't have optional parameters, eg. ReceiveAsync() and ReceiveAsync(TimeSpan operationTimeout). So I am not sure if we want to introduce this.

Reply:
Got it.  To be consistent then, that convention is appropriate.  In the case of ReceiveAsync() its using the OperationTimeout value configured on the connection string or specified when the connection was created and technically is a timeout for a single operation v/s Unregister* which is allowing multiple handlers to complete execution as well as interact over the service bus connection (to complete/abandon messages).  It may be ok to repurpose this.OperationTimeout but if we want a different default, I'd suggest just creating a new configuration element that has a default value. This allows consumers to configure it in a pass through way with managed services like Functions (as is the case for most of the service bus sdk configuration).